bridge/client/app/_components/ktb-heatmap/ktb-heatmap-utils.ts

```diff
@@ -1,62 +1,87 @@
 import { GroupedDataPoints, IDataPoint } from '../../_interfaces/heatmap';
 
 export function createGroupedDataPoints(data: IDataPoint[]): GroupedDataPoints {
-  setUniqueHeaders(data, 'xElement', 'yElement');
-  setUniqueHeaders(data, 'yElement', 'xElement');
+  setUniqueHeaders(data);
   return data.reduce((groupedData: GroupedDataPoints, dataPoint) => {
     groupedData[dataPoint.yElement] ||= [];
     groupedData[dataPoint.yElement].push(dataPoint);
     return groupedData;
   }, {});
 }
 
+function setUniqueHeaders(data: IDataPoint[]): void {
+  setUniqueYAxis(data);
+  setUniqueXAxis(data);
+}
+
 export const getLimitedYElements = (yElements: string[], limit: number): string[] => {
   if (yElements.length <= limit) {
     return yElements;
   }
   return yElements.slice(yElements.length - limit, yElements.length);
 };
 
-export function setUniqueHeaders(dataPoints: IDataPoint[], key: 'xElement', compare: 'yElement'): void;
-export function setUniqueHeaders(dataPoints: IDataPoint[], key: 'yElement', compare: 'xElement'): void;
-export function setUniqueHeaders(
-  dataPoints: IDataPoint[],
-  key: 'yElement' | 'xElement',
-  compare: 'xElement' | 'yElement'
-): void {
-  const duplicatesDict: { [key: string]: { [compare: string]: number } } = {};
+export function setUniqueYAxis(dataPoints: IDataPoint[]): void {
+  // if it is the y-axis compare if there is the same SLI name with the same identifier
+  const duplicatesDict: { [identifier: string]: { [compare: string]: number } } = {};
+  const hasDuplicates: { [yElement: string]: boolean | undefined } = {};
   dataPoints.forEach((dataPoint, index) => {
-    let uniqueHeader = dataPoint[key];
-    const compareWith = dataPoint[compare];
+    let displayValue = dataPoint.yElement;
+    const identifier = dataPoint.identifier;
     let foundIndex;
-    while (
+    if (
       (foundIndex = dataPoints.findIndex(
         // eslint-disable-next-line @typescript-eslint/no-loop-func
-        (dt) => dt[key] === uniqueHeader && dt[compare] === compareWith
+        (dt) => dt.yElement === displayValue && dt.identifier === identifier
       )) < index &&
       foundIndex !== -1
     ) {
```

Comment:
Please refactor this IF. Its very complex, even though you need a lint rule exception. Always name complex IFs such as:
```ts
const condition = ...
if (condition)

or

if (checkCondition(...))
```

Reply:
done